controllers/awscluster_controller.go

```diff
@@ -78,6 +78,7 @@ func (r *AWSClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reter
 		Logger:     log,
 		Cluster:    cluster,
 		AWSCluster: awsCluster,
+		Recorder:   r.Recorder,
```

Comment:
We've usually suggested to use the global recorder instead of a local one, https://github.com/kubernetes-sigs/cluster-api/blob/master/util/record/recorder.go

Reply:
Let's open an issue if we don't have one to get rid of the local one?

Reply:
I pointed it out because it seems this PR is adding the local one, which we probably don't want to

Reply:
Roger... multitasking fail

Reply:
I do however need to a fresh recorder for each test case. In fact, I was extending what was already there, threading the recorder from https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1349/files#diff-a4c89adfffbff9126a6d9d4d96531799L101 into where it needs to go to be able to test the receive of the credential provider chain error.

If there's another refactor that works better, let me know.